perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -0,0 +1,162 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <sstream>
+#include "sim86_shared.h"
+#pragma comment (lib, "sim86_shared_debug.lib")
+#define ArrayCount(Array) (sizeof(Array) / sizeof((Array)[0]))
+
+s16 Registers[15] = {};
```

Comment:
Agreed, fixed here dfc8614128adfd26955f8f54f4e2859e6bff3b66

Reply:
You should use `constexpr` instead of `#define`. There are several differences, but the most important one is that the compiler will know it. Unlike defines where the compiler doesn't know about defines, only the preprocessor does. So if there's an issue with your constant your compiler will not help you if it's just a define, because he doesn't know about it, the preprocessor will have inlined it everywhere. If it's a constexpr, he will tell you about it explicitly, giving you its name. As a general rule, you should always make it easy for the compiler to help you report clear errors as early as possible.

Reply:
Ha! I didn't realize that about #define. Thanks! Fixed here 60b9aa3b1068b46cfba64a9b858381b4c30e9fd9